### ISSUE TITLE
Help port more of eco-system to Scala 2.13

### ISSUE THREAD

**Issue by adriaanm** [scala/scala-dev#563]
We at Lightbend are seeking to collaborate with OSS project maintainers to help port projects from 2.12 to 2.13. This ticket serves to coordinate the effort -- please comment here with any projects you'd like to nominate (including your own ;-)).

**Comment by jvican** [comment]
Migrating anything that is an sbt and Zinc dependency would help so that we can publish a 2.13 version of the incremental compiler before the release of the RC. We're [currently blocked by](https://github.com/sbt/zinc/pull/592):
  
- [x] scalatest
- [x]  sbt-io
- [ ]  sbt-util-logging
- [x]  scalacheck

**Comment by sjrd** [comment]
I'd like to nominate [scala-parallel-collections](https://github.com/scala/scala-parallel-collections). ;)

**Comment by adriaanm** [comment]
Haha, of course the parallel collections were already on the list :-)

**Comment by SethTisue** [comment]
as usual, https://github.com/scala/make-release-notes/blob/2.13.x/projects-2.13.md lists published projects and has links to open tickets for projects that haven't published yet

**Comment by dsilvasc** [comment]
Are [macwire](https://github.com/adamw/macwire), [com.softwaremill.common:tagging](https://github.com/softwaremill/scala-common), [circe](https://github.com/circe/circe), [circe-derivation](https://github.com/circe/circe-derivation), twilio's [guardrail](https://github.com/twilio/guardrail), [sttp](https://github.com/softwaremill/sttp), [gradle-scoverage](https://github.com/scoverage/gradle-scoverage), [gradle-scalastyle-plugin](https://github.com/ngbinh/gradle-scalastyle-plugin), [wartremover](https://github.com/wartremover/wartremover), [sangria](https://sangria-graphql.org/), and [gatling](https://github.com/gatling/gatling) already tracked?

**Comment by martijnhoekstra** [comment]
https://github.com/twitter/util could probably use some help, and is often a deep transitive dependency

**Comment by martijnhoekstra** [comment]
Also, https://github.com/apache/spark

**Comment by da-liii** [comment]
Any guide to port a project from 2.12 to 2.13 ?

I'm willing to work on Apache Spark.

**Comment by da-liii** [comment]
Get the dependents:
```
$ ./build/mvn -Phive-thriftserver -DskipTests package
$ ls assembly/target/scala-2.12/jars  | grep _2.12 | grep -v spark
```

Well, Apache Spark depends on ...

+ [ ] breeze-macros_2.12-0.13.2.jar
+ [ ] breeze_2.12-0.13.2.jar
+ [ ] chill_2.12-0.9.3.jar
+ [x] jackson-module-scala_2.12-2.9.6.jar
+ [x] json4s-ast_2.12-3.5.3.jar (3.6.1)
+ [x] json4s-core_2.12-3.5.3.jar
+ [x] json4s-jackson_2.12-3.5.3.jar
+ [x] json4s-scalap_2.12-3.5.3.jar
+ [x] machinist_2.12-0.6.1.jar (0.6.6)
+ [x] macro-compat_2.12-1.1.1.jar (1.1.1)
+ [x] shapeless_2.12-2.3.2.jar (2.3.3)
+ [ ] spire-macros_2.12-0.13.0.jar
+ [ ] spire_2.12-0.13.0.jar


## Related Issues
+ https://github.com/json4s/json4s/issues/541
+ https://github.com/FasterXML/jackson-module-scala/issues/391
+ https://github.com/scalanlp/breeze/issues/738
+ https://github.com/non/spire/issues/742

## Check on M5
+ [x] https://oss.sonatype.org/content/repositories/releases/com/fasterxml/jackson/module/
+ [x] https://oss.sonatype.org/content/repositories/releases/org/json4s/
+ [x] https://oss.sonatype.org/content/repositories/releases/org/typelevel/macro-compat_2.13.0-M5/
+ [x] https://search.maven.org/search?q=a:machinist_2.13.0-M5
+ [x] https://search.maven.org/search?q=shapeless_2.13.0-M5
+ [ ] https://search.maven.org/search?q=spire_2.13.0-M5
+ [ ] https://search.maven.org/search?q=chill_2.13.0-M5

## The Spark Jira
https://issues.apache.org/jira/browse/SPARK-25075

**Comment by jvican** [comment]
@sadhen's suggestions are great. I would prioritize those deps to make sure we don't repeat the same mistake with Apache Spark in 2.13. Even if it's not going to be released this year, the closer we are from having a working 2.13 Apache Spark port available, the better.
  
(I don't know how much does Spark use Scala collections, but if it does I also think it would be a good real-world use case to assess the performance of the new collections.)

**Comment by szeiger** [comment]
If you're porting code to 2.13, check out the [presentation](https://www.youtube.com/watch?v=pHHKUKubs1Q&list=PLLMLOC3WM2r6Z6dIUi01-XXI4FmDrAl20&index=17) on this topic by @julienrf and myself from ScalaDays. For the new collections library we have an updated [documentation](https://docs.scala-lang.org/overviews/core/architecture-of-scala-213-collections.html) on scala-lang. For cross-building on 2.12 and 2.13 you may want to use the  [collection-compat](https://github.com/scala/scala-collection-compat/) library.

**Comment by smarter** [comment]
It looks like https://github.com/scala/bug/issues/11030 is preventing fs2 and probably other projects from compiling with 2.13: https://github.com/functional-streams-for-scala/fs2/issues/1232, so fixing it would definitely count as helping the ecosystem migrate :).

**Comment by dsilvasc** [comment]
Gatling request: https://github.com/gatling/gatling/issues/3566

**Comment by slandelle** [comment]
Regarding Gatling, we depend on (listing only libs that haven't started supporting 2.13):
* [fastring](https://github.com/Atry/fastring)
* [spire](https://github.com/non/spire) (macros)
* [boopickle](https://github.com/suzaku-io/boopickle)
* zinc
* [quicklens](https://github.com/adamw/quicklens)

**Comment by Sciss** [comment]
We have a ticket for scala-swing now:

https://github.com/scala/scala-swing/issues/84

**Comment by Rich2** [comment]
I'd certainly like to see help for Scala-Native. Getting to 2.12 seems to be the big challenge for them. Hopefully 2.12 => 2.13 will be relatively straightforward.

**Comment by TimMoore** [comment]
https://repo1.maven.org/maven2/com/fasterxml/jackson/module/jackson-module-scala_2.13.0-M5/2.9.8/jackson-module-scala_2.13.0-M5-2.9.8.jar is now available

**Comment by SethTisue** [comment]
volunteer to port [boopickle](https://github.com/suzaku-io/boopickle/issues/113)? it's blocking http4s and gatling (and perhaps other projects)

**Comment by skonto** [comment]
For spark here is the [dependency tree](https://gist.github.com/skonto/b046f2e350106754abd799833fe945f1) (no profile used, we need to verify with profiles to see what they bring in). Verified @sadhen's list.  Pending issues:
* [Spire](https://github.com/typelevel/spire/issues/788).
* [Chill](https://github.com/twitter/chill/issues/316).
* [Breeze](https://github.com/scalanlp/breeze/issues/738) blocked by the Spire issue.

**Comment by szeiger** [comment]
Spire depends on Algebra which doesn't have a 2.13 release yet. WIP here: https://github.com/typelevel/algebra/pull/226

**Comment by adriaanm** [comment]
See also https://github.com/scalacommunitybuild/algebra/tree/community-build-2.13 (and generally, the same pattern of branch name in that org for many projects)

**Comment by adriaanm** [comment]
FYI, 2.13.0-RC2 is tagged and on Maven Central! We'll announce after the weekend.

**Comment by adriaanm** [comment]
Looks like [spire](https://github.com/typelevel/spire/pull/796) and [algebra](https://github.com/typelevel/algebra/pull/228) are up to date with 2.13.0-RC1, and I don't expect RC2 to be an issue for them!

**Comment by adriaanm** [comment]
For chill & breeze, we should start with twitter-util and scrooge, which are already in the [community build](https://github.com/scala/community-builds/blob/2.13.x/configs/project-refs.conf#L167). To see the errors, grep for `[twitter-util] [error]` in https://scala-ci.typesafe.com/job/scala-2.13.x-integrate-community-build/2208/consoleText.

**Comment by martijnhoekstra** [comment]
I have a WIP for twitter-util at https://github.com/twitter/util/pull/248 -- would particularly appreciate ideas for https://github.com/twitter/util/pull/248/files#r279409022

**Comment by dsilvasc** [comment]
Looks like [monadless](https://github.com/monadless/monadless) isn't in the community build.  There's a question about 2.13 support here: https://github.com/monadless/monadless/issues/9

**Comment by Philippus** [comment]
Any reason scalamock is not pointing to https://github.com/paulbutcher/ScalaMock ?

**Comment by SethTisue** [comment]
over at https://github.com/scala/make-release-notes/blob/2.13.x/projects-2.13.md I'm not really doing any more maintenance on the list of what *is* published for 2.13 (except merging PRs on it), but I am definitely maintaining the list of "please publish for 2.13" tickets at https://github.com/scala/make-release-notes/blob/2.13.x/projects-2.13.md#pending, so that's a place to look for projects to help upgrade

**Comment by ayoub-benali** [comment]
> For spark here is the [dependency tree](https://gist.github.com/skonto/b046f2e350106754abd799833fe945f1) (no profile used, we need to verify with profiles to see what they bring in). Verified @sadhen's list. Pending issues:
> 
> * [Spire](https://github.com/typelevel/spire/issues/788).
> * [Chill](https://github.com/twitter/chill/issues/316).
> * [Breeze](https://github.com/scalanlp/breeze/issues/738) blocked by the Spire issue.

Spire has a [0.17.0-M1](https://github.com/typelevel/spire/releases/tag/v0.17.0-M1) release which is available for Scala 2.13

**Comment by SethTisue** [comment]
this is an open-ended/never-ending task, but I think this ticket has now served its purpose. thanks all!

**Comment by SethTisue** [comment]
I'm continuing to maintain https://github.com/scala/make-release-notes/blob/2.13.x/projects-2.13.md — it lists projects that are known not to support 2.13 yet

**Comment by rafalmag** [comment]
I would like to nominate [zinc](https://github.com/sbt/zinc/) (see sbt/zinc#697) , which seems to block [gatling](https://github.com/gatling/gatling) (see gatling/gatling#3566).

**Comment by SethTisue** [comment]
@rafalmag I added the link to https://github.com/scala/make-release-notes/blob/2.13.x/projects-2.13.md
